src/ChatPage.cpp

```diff
@@ -446,6 +465,31 @@ ChatPage::ChatPage(QSharedPointer<UserSettings> userSettings, QWidget *parent)
                                   roomid, filename, encryptedFile, url, mime, dsize);
                 });
 
+        connect(text_input_, &TextInputWidget::callButtonPress, this, [this]() {
+                if (callManager_.onActiveCall()) {
+                        callManager_.hangUp();
+                } else {
+                        if (cache::singleRoomInfo(current_room_.toStdString()).member_count != 2) {
+                                showNotification("Voice/Video calls are limited to 1:1 rooms");
+                        } else {
+                                std::vector<RoomMember> members(
+                                  cache::getMembers(current_room_.toStdString()));
+                                const RoomMember &callee =
+                                  members.front().user_id == utils::localUser() ? members.back()
+                                                                                : members.front();
+                                auto dialog =
+                                  new dialogs::PlaceCall(callee.user_id, callee.display_name, this);
+                                connect(dialog, &dialogs::PlaceCall::voice, this, [this]() {
+                                        callManager_.sendInvite(current_room_);
+                                });
+                                connect(dialog, &dialogs::PlaceCall::video, this, [this]() {
+                                        showNotification("Video calls not yet implemented");
+                                });
+                                dialog->show();
```

Comment:
At least on my system this dialog doesn't seem to be centered on Nheko correctly, you may want to use something like `utils::centerWidget`.